File: XPile 3-17/Programs/test_with.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def get_resource():
    return 42

def main():
    r = get_resource()
    try:
        print(r)
    finally:
        if hasattr(r, 'close'): r.close()
    val = get_resource()
    try:
        doubled = (val * 2)
        print(doubled)
    finally:
        if hasattr(val, 'close'): val.close()
    print(1)
    return 0


if __name__ == "__main__":
    main()

File: XPile 3-17/Programs/test_with.js

function get_resource() {
    return 42;
}

function main() {
    let r = get_resource();
    try {
        console.log(r);
    } finally {
        if (r && typeof r.close === 'function') r.close();
    }
    let val = get_resource();
    try {
        let doubled = (val * 2);
        console.log(doubled);
    } finally {
        if (val && typeof val.close === 'function') val.close();
    }
    console.log(1);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
